assets/js/modules/analytics/common/account-create.js

```diff
@@ -0,0 +1,163 @@
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, Fragment, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../../../components/button';
+import ProgressBar from '../../../components/progress-bar';
+import { trackEvent } from '../../../util';
+import TimezoneSelect from './timzezone-select';
+import AccountField from './account-field';
+import PropertyField from './property-field';
+import ProfileField from './profile-field';
+
+// import Data from 'googlesitekit-data';
+// const { dispatch, select } = Data;
+
+const AccountCreate = () => {
+	// const { createAccount } = dispatch( STORE_NAME );
+	// const isDoingCreateAccount = select( STORE_NAME ).isDoingCreateAccount();
+	const isDoingCreateAccount = false;
+	const { createAccount } = () => {};
+	const { siteName, siteURL, timezone: tz, errorcode } = global.googlesitekit.admin;
+
+	// Handle expected provisioning flow error codes.
+	let errorMessage = false;
+	switch ( errorcode ) {
+		case 'user_cancel':
+			errorMessage = __( 'The Terms of Service were not accepted.', 'google-site-kit' );
+			break;
+
+		case 'max_accounts_reached':
+			errorMessage = __( 'The Google Analytics account limit has been reached.', 'google-site-kit' );
+			break;
+
+		case 'backend_error':
+			errorMessage = __( 'Unknown service error.', 'google-site-kit' );
+			break;
+	}
+	const [ error, setError ] = useState( errorMessage );
+
+	const handleSubmit = useCallback( ( accountName, propertyName, profileName, timezone ) => {
+		trackEvent( 'analytics_setup', 'new_account_setup_clicked' );
+		createAccount( {
+			accountName,
+			propertyName,
+			profileName,
+			timezone,
+		} ).then( ( e ) => {
+			const { error: err } = e.payload;
+			if ( err ) {
+				setError( err.message ? err.message : __( 'Unknown error.', 'google-site-kit' ) );
+			}
+		} );
+	} );
+
+	const [ accountName, setAccountName ] = useState( siteName );
+	const [ propertyName, setPropertyName ] = useState( siteURL );
+	const [ profileName, setProfileName ] = useState( __( 'All website traffic', 'google-site-kit' ) );
+	const [ validationIssues, setValidationIssues ] = useState( {} );
+	const [ timezone, setTimezone ] = useState( tz );
+
+	// Disable the submit button if there are validation errors, and while submission is in progress.
+	const buttonDisabled = validationIssues.accountName || validationIssues.propertyName || validationIssues.profileName || isDoingCreateAccount;
+
+	return (
+		<Fragment>
+			<div className="googlesitekit-setup-module">
+				<div className="mdc-layout-grid__inner">
+					<div className="mdc-layout-grid__cell--span-12">
+						<h2>
+							{ __( 'Create new Analytics account', 'google-site-kit' ) }
+						</h2>
+						{
+							error &&
+							<div className="error">
+								<p>
+									{ error }
+								</p>
+							</div>
+						}
```

Comment:
great, I will for that. This was a temporary solution that needs improvement.

Reply:
Updated in https://github.com/google/site-kit-wp/pull/1389/commits/d445f69708dc5f8634749d78613eb4680eb3b509.

Reply:
Do you mean using `GenericError` or some other component? (/assets/js/components/notifications/generic-error.js)

Reply:
@aaemnnosttv ^^

Reply:
There is an `ErrorNotice` component specific to Analytics, in `js/modules/analytics/common`. That automatically uses `getError`, so based on my comments above I think you wouldn't even need to call `getError` at all in this component, and rather just render the `ErrorNotice` component.

Reply:
Since the callback handler is displayed on analytics in an iframe, it makes more sense to show any error messaging there. I am removing the error handling from this component.

Reply:
removed in https://github.com/google/site-kit-wp/pull/1389/commits/d007a17391c76d7f19b81caa09f55fe808b483c9